src/templates/finops-hub/modules/dataFactory.bicep

```diff
@@ -4863,6 +4939,401 @@ resource pipeline_RerunETL 'Microsoft.DataFactory/factories/pipelines@2018-06-01
+            type: 'DatasetReference'
+            parameters: {
+              fileName: 'Recommendations.json'
+              folderPath: '${configContainerName}/queries'
+            }
+          }
+          firstRowOnly: true
+        }
+      }
+      { // Iterate Queries
+        name: 'Iterate Queries'
+        type: 'ForEach'
+        dependsOn: [
+          {
+            activity: 'Set Error Counter'
+            dependencyConditions: [
+              'Succeeded'
+            ]
+          }
+          {
+            activity: 'Set Blob Base Path'
+            dependencyConditions: [
+              'Succeeded'
+            ]
+          }
+          {
+            activity: 'Load Schema Mappings'
+            dependencyConditions: [
+              'Succeeded'
+            ]
+          }
+          {
+            activity: 'Load Queries'
+            dependencyConditions: [
+              'Succeeded'
+            ]
+          }
+        ]
+        userProperties: []
+        typeProperties: {
+          items: {
+            value: '@activity(\'Load Queries\').output.firstRow.queries'
+            type: 'Expression'
+          }
+          isSequential: true
+          activities: [
+            { name: 'Execute Query'
+              type: 'Copy'
+              dependsOn: [
+              ]
+              policy: {
+                timeout: '0.00:10:00'
+                retry: 0
+                retryIntervalInSeconds: 60
+                secureOutput: false
+                secureInput: false
+              }
+              userProperties: []
+              typeProperties: {
+                source: {
+                  type: 'RestSource'
+                  httpRequestTimeout: '00:02:00'
+                  requestInterval: '00.00:00:00.050'
+                  requestMethod: 'POST'
+                  requestBody: {
+                    value: '@concat(\'{ "query": "\', item().query, \' | extend x_SourceName=\\"\', item().source, \'\\", x_SourceType=\\"\', item().type, \'\\", x_SourceProvider=\\"\', item().provider, \'\\", x_SourceVersion=\\"\', item().version, \'\\"" }\')'
+                    type: 'Expression'
+                  }
+                  additionalHeaders: {
+                    'Content-Type': 'application/json'
+                  }
+                }
+                sink: {
+                  type: 'ParquetSink'
+                  storeSettings: {
+                    type: 'AzureBlobFSWriteSettings'
+                  }
+                  formatSettings: {
+                    type: 'ParquetWriteSettings'
+                    fileExtension: '.parquet'
+                  }
+                }
+                enableStaging: false
+                translator: {
+                  value: '@activity(\'Load Schema Mappings\').output.firstRow.translator'
+                  type: 'Expression'
+                }
+              }
+              inputs: [
+                {
+                  referenceName: 'resourcegraph'
```

Comment:
This should be dynamic based on the data source to support other targets. It's fine if we don't support them now, but we should at least be prepared so we don't have a bunch of cleanup work later.

Also, use the `dataset_resourcegraph.name` variable reference instead of hard-coding.

Reply:
Changed to refer to  `dataset_resourcegraph.name` instead.

Regarding dynamic data sets based on the data source, I agree this will be needed sooner or later. However, this ADF activity was specifically designed for ARG queries and making it generic at this moment would make it much more complex. I prefer to keep it simple at this moment and be prepared to make it more generic when the need arises.

Reply:
With the pipeline refactoring, I ended up implementing what you suggested. It hurt but the result looks good :-)